Odongo006/Rails_blog_App
# Add devise

## Description
This pull request addresses the project requirements by implementing the necessary changes and additions to enhance user authentication and registration using Devise. The modifications include removing the current_user method from ApplicationController as Devise provides its own method. Additionally, Devise is installed and configured to enable registration, login, password reset, and email confirmation functionalities. The views for registration and login are also modified to align with the provided wireframes and incorporate your custom styling.

## Changes Made

1. Removed the current_user method from ApplicationController.
2. Installed and configured Devise gem.
3. Implemented user registration feature:

> Users can register with a unique email and password combination.
> Hashed passwords are securely stored in the database.
> Confirmation email is sent to the user for email verification.

4. Implemented user login feature:

> Users can log in using their registered email and password.

5. Implemented password reset feature:

> Users can request a password reset via email.

6. Modified Devise views for registration and login:

> Updated the views to match the wireframes provided in the Sneak Peek.
> Applied your custom styling to enhance the user interface.

## How to Test

1. Ensure Devise is properly installed and the gem dependencies are up to date.
2. Migrate the database to incorporate the new user table and columns.
3. Start the application server.
4. Access the registration page and create a new user account.
5. Verify the registration confirmation email and follow the instructions to confirm the email address.
6. Log in using the registered email and password.
7. Test the password reset feature by requesting a password reset via email and following the instructions.
8. Validate the modified views for registration and login to ensure they match the provided wireframes and styling.

Please review the changes and test the functionality thoroughly. Your feedback is greatly appreciated.
# N/B: I would love to inform you that I worked on this project alone, since I have no partner to pair with, as it is evident below::
<img width="1281" alt="collaboration" src="https://github.com/Odongo006/Rails_blog_App/assets/51127665/4222ea91-ea12-4e8c-bd86-5ed364d7a8f7">
